WCFDataService/Client/System/Data/Services/Client/MaterializeFromAtom.cs

```diff
@@ -69,6 +69,11 @@ internal class MaterializeAtom : IDisposable, IEnumerable, IEnumerator
         private bool moved;
 
         /// <summary>
+        /// The ODataResponseMessage to dispose when we are done.
+        /// </summary>
+        private IODataResponseMessage responseMessage;
```

Comment:
Seems MaterializeFromAtom.Dispose will not be called explicitly. One of our cases is failed because the response message is not disposed. What about like handling $count, we use a "finally" of every caller of ExecuteQuery() to dispose response message?